All Views created & ready for integration. Some changes made to Models.

grahamhome opened the pull request:
Resolves #6 

We had to make some changes to the Models classes to ensure we have the correct data to display in our Views. @Ethan7 should review the changes to Models so he knows how to use them in his database-access methods.

grahamhome committed:
Removed a class which appears to have been accidentally committed

grahamhome committed:
Merge branch 'master' into view

grahamhome committed:
Merge branch 'master' into view

Commit:
Updated Store Management View

Used tabs instead of buttons to display shipments view, purchases view and inventory view.

grahamhome committed:
Updated views & models to support adding items to cart with a specified quantity

Commit:
Created the StoreSelectionViewController to add on click action functionality for when a store is clicked. Controller displays store information sucessfully. Needs to be integrated when controller tier interaction is implemented.

grahamhome committed:
Updated CartViewController to support editing item quantity

grahamhome committed:
Merge branch 'view' of https://github.com/atla5/csci320_rc into view

Commit:
Uploading New Views

Created SelectVenforView to display the list of vendors and VendorShipmentView to query the list of shipments that a specific vendor did

grahamhome committed:
Bugfixes

Commit:
Created the SelectVendorListController, edited the session data and ActivityManager classes to fit the change, fixed the data display in the CustomerViewCController class

grahamhome committed:
Merge branch 'view' of https://github.com/atla5/csci320_rc into view

grahamhome committed:
Shopping use case is complete and ready for integration!

grahamhome committed:
Added button-press logic to all buttons in MainView

grahamhome committed:
Added vendor shipment details screen and small improvements to existing screens

Commit:
Created and linked the classes and fxml files required for the vendors use cases.

grahamhome committed:
Merge branch 'view' of https://github.com/atla5/csci320_rc into view

grahamhome committed:
Linked vendor shipment details screen to vendor shipments screen

atla5 committed:
dataloader: product and store import

atla5 committed:
dataloader: additional helpers for other tables

atla5 committed:
dataloader: database files loaded with sample data

atla5 committed:
dataloader: [refactor] mv DataLoader controllers/, extract insert, proxy preview

atla5 committed:
dataloader: additional insert helpers to other proxies

atla5 committed:
dataloader: add some TODOs regarding db tables/sample_data

atla5 committed:
dataloader: undo accidental commits

atla5 committed:
dataloader: update database with changes

grahamhome committed:
Cleaned up models

grahamhome committed:
Cleaned up models

grahamhome committed:
Merge branch 'view' of https://github.com/atla5/csci320_rc into view

grahamhome committed:
Merge branch 'view' of https://github.com/atla5/csci320_rc into view

atla5 commented on src/main/java/com/rainforestcommerce/rcdb/models/ProductQuantityPrice.java:
```diff
@@ -0,0 +1,51 @@
+    private int quantity;
+
+
+    public ProductQuantityPrice(long purchaseId, long productId, int unit_price, int quantity, Product product){
```
I'd rename this to something more semantic relating to the store's inventory. e.g. `StoreInventoryItem`.

atla5 commented on src/main/java/com/rainforestcommerce/rcdb/views/V_ShipmentsRequestViewController.java:
```diff
@@ -0,0 +1,80 @@
+ * @author Abdullah Al Hamoud <aaa1008@g.rit.edu>
+ *
+ */
+public class V_ShipmentsRequestViewController {
```
why shorten `Vendor` here? (the other one is distinct since it's `VendorShipmentDetails`

atla5 commented on src/main/java/com/rainforestcommerce/rcdb/views/V_ShipmentsRequestViewController.java:
```diff
@@ -0,0 +1,80 @@
+
+        // TEST CODE - remove when DB access is implemented
+
+        // End test code
```
just a note about the comments here. i'd ideally remove them and add tests if desired.

grahamhome committed:
Bugfix

grahamhome committed:
Merge branch 'view' of https://github.com/atla5/csci320_rc into view

Commit:
Fixed bug

Fixed a null pointer exception from FXML loader.

Ethan7 committed:
Modified the purchasing products function accordingly.

grahamhome committed:
Merge branch 'master' into dataloader

grahamhome committed:
Merge branch 'master' into dataloader

grahamhome committed:
Merge branch 'view' into controller

Commit:
UI Button Functionality

All button functionality should work in the user interface.

Ethan7 committed:
Fixed a bunch of null pointer exceptions, changed id's to long's, fixed formatting on insertion statements, fixed sql statements to match documentation names, and a few other things.

Ethan7 committed:
Added table creation code, fixed references to DataLoader, added logger to DataLoader, and fixed more SQL code.

Commit:
Added table generation for management view shipments and purchases

grahamhome committed:
Merge branch 'view' of https://github.com/atla5/csci320_rc into view

grahamhome committed:
Added correct cost calculation function to StorePurchase

grahamhome committed:
Added call to CustomerProxy from CustomerSelectionVIew

grahamhome committed:
Merge branch 'controller' into view

grahamhome committed:
Bugfix

Ethan7 committed:
Changed ID longs to bigint in SQL code

grahamhome committed:
Deleted defunct databases, added code to call CustomerProxy

grahamhome committed:
Merge branch 'Controller' into view

grahamhome committed:
Revert "Deleted defunct databases, added code to call CustomerProxy"

This reverts commit de90faeb22fefc6926d39e7ac4e921528caa9fae.

grahamhome committed:
Pre-made DB connection working

Ethan7 committed:
Fixed Customer problems.

Commit:
Small changes to fix proxy errors

Commit:
Merge remote-tracking branch 'origin/view' into view

Ethan7 committed:
Fixed Customer problems.

Commit:
Merge remote-tracking branch 'origin/view' into view

atla5 committed:
dataloader: storePurchase.online[-store], +db update

atla5 committed:
Merge branch 'dataloader' of https://github.com/atla5/csci320_rc into dataloader

grahamhome committed:
Merge branch 'dataloader' into view

atla5 committed:
dataloader: [refactor] make dataDirectory path generic and absolute

Ethan7 committed:
Fixed Store Problems

Ethan7 committed:
Merge branch 'view' of https://github.com/atla5/csci320_rc into view

atla5 committed:
dataloader: switch from 'System.*.println' to 'LOGGER'

grahamhome committed:
Bugfix for Store Selection View

Ethan7 committed:
Fixed Problem with always creating the database.

Ethan7 committed:
Fixed Problems with always creating the database again.

Ethan7 committed:
Merge branch 'view' of https://github.com/atla5/csci320_rc into view

atla5 committed:
p3data: update dataloader with shipment, vendor loaders

atla5 committed:
p3data: update db with shipments, vendors

atla5 committed:
p3data: [refactor] go back to using LOGGER instead of System.out

it's much easier to work with System.out output when i'm running the output commands manually.

Ethan7 committed:
Fixed getStores to includes inventory of products

Ethan7 committed:
In the process of fixing the rest of getStores.

grahamhome committed:
Merge pull request #17 from atla5/p3data

p3data: Create and populate 'Shipment*', 'Vendor*' tables in db, dataloader

grahamhome committed:
Merge branch 'dataloader' into view

Ethan7 committed:
Fixed last of getStore code.

grahamhome committed:
Fixed vendor table name reference

Ethan7 committed:
Merge branch 'view' of https://github.com/atla5/csci320_rc into view

grahamhome committed:
Fixed database table name references

Ethan7 committed:
Fixed final last final penultimate bit of getStores.

Ethan7 committed:
Fixed final last final penultimate extra super finito bit of getStores.

Ethan7 committed:
Continuous fixing and testing: parse date for Shipments.

grahamhome committed:
Bugfixes

Ethan7 committed:
Continuous fixing and testing: fixed array loop for getStores

grahamhome committed:
Bugfixes

grahamhome committed:
Added method call to place a purchase from the Cart view

Ethan7 committed:
Fixed the purchasing option for the view and the controller.

Ethan7 committed:
Added that purchases from the view side automatically set the date of the purchase to the current date. And all product purchases within the store purchase are properly added to the database.

grahamhome committed:
Added feature to prevent unfulfillable orders from being placed; fixed bug with StoreView not being updated when loaded

grahamhome committed:
Merge branch 'view' of https://github.com/atla5/csci320_rc into view

Commit:
Updated the UI to show the purchase points to the user. Added functionality to calculate purchase points in the StorePurchase model class.

Ethan7 committed:
Fixed Purchase code to subtract from inventory.

Ethan7 committed:
Merge branch 'view' of https://github.com/atla5/csci320_rc into view

Ethan7 committed:
Added points to customer constructor as well as database grab.

grahamhome committed:
Bugfixes and code cleanup

grahamhome committed:
Merge branch 'view' of https://github.com/atla5/csci320_rc into view

grahamhome committed:
Added view and controller for Enterprise Insights use cases - no functionality in controller yet
